ShopperInsights/src/main/java/com/braintreepayments/api/ShopperInsightsClient.kt

```diff
@@ -47,4 +48,36 @@ class ShopperInsightsClient @VisibleForTesting internal constructor(
             )
         )
     }
+
+    /**
+     * Call this method when the PayPal button has been successfully displayed to the buyer.
+     * This method sends analytics to help improve the Shopper Insights feature experience.
+     */
+    fun sendPayPalPresentedEvent() {
```

Comment:
Do we have a way to ensure that merchants invoke these analytics functions? Not sure if there's a way when the merchant owns the PayPal/Venmo button 🤔

Reply:
https://github.com/braintree/braintree_ios/pull/1162
I pretty much emulated the iOS PR.

> Do we have a way to ensure that merchants invoke these analytics functions?

I had the same question, but pretty sure we are at their mercy. I don't see how we can do client side analytics when they own the UI.

Reply:
➕ 1 yeah we are adding these in hopes that merchants actually send the events but can't really force merchants to do anything with their own UI like Sai said.